frontend/app/src/components/account-management/create-account/premium/CreateAccountPremium.vue

```diff
@@ -0,0 +1,101 @@
+<script lang="ts" setup>
+import { type Ref } from 'vue';
+import { type PremiumSetup } from '@/types/login';
+import CreateAccountPremiumForm from '@/components/account-management/create-account/premium/CreateAccountPremiumForm.vue';
+
+const props = defineProps<{
+  loading: boolean;
+  form: PremiumSetup;
+  premiumEnabled: boolean;
+}>();
+
+const emit = defineEmits<{
+  (e: 'back'): void;
+  (e: 'next'): void;
+  (e: 'update:form', form: PremiumSetup): void;
+  (e: 'update:premium-enabled', enabled: boolean): void;
+}>();
+
+const { premiumEnabled, form } = toRefs(props);
+
+const { t } = useI18n();
+
+const store = useMainStore();
+const { newUser } = toRefs(store);
+
+const valid: Ref<boolean> = ref(false);
+
+const premiumSelectionButtons = [
```

Comment:
wouldn't `computed` make sense here for future purposes? `t` should be reactive in the next version